### DISCUSSION TITLE
Icon labels and Hint labels are not displayed

### DISCUSSION THREAD

**Discussion by sumedhkumar96** [OnionUI/Onion#1634] [Support]
### Checklist

- [X] I am reporting a bug in the Onion OS - Any issues with the closed source frontend ([MainUI](https://github.com/OnionUI/MainUI-issues)), or [RetroArch](https://github.com/libretro/RetroArch) or any of the cores should be directed to their respective repositories
- [X] I have reviewed the [Onion Docs](https://onionui.github.io/docs) and did not find the answer
- [X] I have searched the existing (open and closed) issues to ensure I do not log a duplicate

### Onions OS Version

4.3.1-1

### Miyoo Firmware Version

202306282128

### Provide a clear and concise description of the issue

I recently updated my Miyoo Mini Plus Onion OS 4.3.0 to 4.3.1-1 using Onion Desktop Tools 0.0.9. I also saw the new themes on the official page so I downloaded all the themes on the whole repository (I wanted to try it all) and copied the extracted folders onto the Themes directory. I then inserted the SD card into the device and booted it up and the update was successful. 

Until this point, I was able to see the Icon labels and Hint labels on the existing theme (Wavedash theme). Next, I went to the Themes app and applied one of them and I noticed that the Icon labels and Hint labels were all gone. I thought it was a theme issue and I switched back to the old theme I had (Wavedash), and I saw that is also broken. Next, I checked the Tweaks app to see if there were any theme overrides but it was all set to default behavior. 

I tried restarting the device, and switching to the default Silky theme, but nothing worked. So, I decided to reinstall the OS. After re-installation the labels re-appeared! I was happy and I changed the theme again, and the labels disappeared again... 

![Capture](https://github.com/user-attachments/assets/3bd6ca0f-7b0b-4c0d-be9c-2683eb799784)
![Capture2](https://github.com/user-attachments/assets/91e654ac-8d92-4816-8e6c-3ed0d30ed021)

### Steps to Reproduce

1. Update the Onion OS from 4.3.0 to 4.3.1-1 using Onion Desktop Tools 0.0.9.
2. Download all the themes on the repository, extract, and place the folders in the Themes directory.
3. Insert the SD card into the device and update the OS.
4. Change the themes, try switching multiple themes. 

### Provide any additional context or information that may help us investigate your issue

I am not sure if any of the themes are messing up the Theme app and later it is affecting all the themes. Just a hunch. There are some themes that still work perfectly. For example, one of my favorite themes, win98 by kyhynngy_oyuur still displays all the labels and it has got perfect icon support especially for Wifi. (most of the themes, even the latest ones, don't have uniform icons everywhere, some of the themes don't have the Wifi icon support. Maybe that's a different bug altogether)

At first, I thought it was all legacy theme issues but this issue is seen on the latest themes as well. So I think, there is something else going on.

![Capture3](https://github.com/user-attachments/assets/4980c646-06cc-4c45-b576-6ccfd37926c5)

**Comment by Aemiii91** [comment]
To fix this, delete `miyoo/app/lang` and reinstall Onion.

**Reply by sumedhkumar96** [reply]
I tried this but it is still the same

**Reply by Aemiii91** [reply]
Have you tried a different SD card? Or reformatting the card?

**Reply by sumedhkumar96** [reply]
I moved to a new SD card in January 2024 when I still had Onion OS 4.3.0. It is of size of 128 GB. However, this behavior was not seen in 4.3.0.

**Comment by Aemiii91** [comment]
Also make sure the two "hide labels" options aren't enabled in **Tweaks -> Appearance -> Theme overrides**

**Reply by sumedhkumar96** [reply]
Yes the overrides are set to theme defaults (-)

**Reply by sumedhkumar96** [reply]
I manually set 'Hide icon label' and 'Hide icon hints' to 'Off' and I see it is working for some themes.

**Reply by sumedhkumar96** [reply]
Some of the latest themes are still missing Icon Labels, even with 'Hide icon label' and 'Hide icon hints' set to 'Off'.

**Reply by sumedhkumar96** [reply]
The behavior is not consistent, now after changing multiple themes, the Icon Labels are not visible but Icon hints are visible. Theme overrides, 'Hide icon label' and 'Hide icon hints' are set to 'Off'.

**Comment by sumedhkumar96** [comment]
Did anybody else find the same issue? Any work-around for this?

**Reply by Raven7738** [reply]
Hi - I’m getting this issue now and wondering if you were able to fix it and what the fix was?

**Reply by sumedhkumar96** [reply]
Nope. I was not able to find any workaround for this yet. This theme happens to work well though: **win98 by kyhynngy_oyuur** 
All the other themes still have the same issue.

**Reply by DekosAnjo** [reply]
> Hi - I’m getting this issue now and wondering if you were able to fix it and what the fix was?

make a clean install, then edit this file App/ThemeSwitcher/launch.sh to

```
#!/bin/sh
echo $0 $*
[ -e /mnt/SDCARD/miyoo/app/.lang ] || cp -r /mnt/SDCARD/miyoo/app/lang /mnt/SDCARD/miyoo/app/.lang
cp -r /mnt/SDCARD/miyoo/app/.lang/. /mnt/SDCARD/miyoo/app/lang/
cd /mnt/SDCARD/.tmp_update
./bin/themeSwitcher
```
